### PR TITLE
Document premium referrals (nitro trials)

### PR DESCRIPTION
**Pull request by hackermondev**

### CODE REVIEW

File: pages/resources/message.mdx [hunk 1]
```diff
@@ -93,6 +93,7 @@ Type `19` and `20` are only available in API v8 and above. In v7 and below, they
...
 | 31     | STAGE_TOPIC                                  | A message sent when a stage channel's topic is changed                    | "{author} changed the Stage topic: **{content}**"                                                                                                                                | true      |
 | 32     | GUILD_APPLICATION_PREMIUM_SUBSCRIPTION       | A message sent when a user purchases an application premium subscription  | "{author} upgraded {application ?? "a deleted application"} to premium for this server!"                                                                                         | true      |
+| 35     | PREMIUM_REFERRAL                             | A message sent when a user gifts a nitro referral                         | "{content}"                                                                                                                                                                      | true      |
```
Comment (dolfies) [inline pages/resources/message.mdx:96]:
```suggestion
| 35     | PREMIUM_REFERRAL                             | A message sent when a user gifts a premium (Nitro) referral              | "{content}"                                                                                                                                                                      | true      |
```

**Review by dolfies — CHANGES_REQUESTED** [review]

**Comment by hackermondev** [comment]
@dolfies recheck new changes 🙏

File: pages/resources/premium-referral.mdx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+
+### Premium Referral Eligibility Object
+
+Used to represent a user's premium referral eligibility.
+
+###### Premium Referral Eligibility Structure
+| Field                 | Type                                                        | Description                                                                                              |
+| --------------------- | ----------------------------------------------------------- | -------------------------------------------------------------------------------------------------------- |
+| referrals_remaining   | number                                                      | Amount of referrals you have remaining                                                                   |
```
Comment (dolfies) [inline pages/resources/premium-referral.mdx:28]:
```suggestion
| referrals_remaining   | number                                                      | The amount of referrals remaining                                                                |
```

File: pages/resources/premium-referral.mdx [hunk 3]
```diff
@@ -0,0 +1,80 @@
+# Premium Referral Resource
+
+Premium referrals, or Nitro trials, allow Nitro subscribers to share up to three 2-week Nitro subscriptions with friends who haven't had an active Nitro subscription in the past 12 months.
+
+### Premium Referral Object
+
+Used to represent a premium referral.
+
```
Comment (dolfies) [inline pages/resources/premium-referral.mdx:8]: Unnecessary
```suggestion
```

File: pages/resources/premium-referral.mdx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+###### Premium Referral Structure
+
+| Field                 | Type                                                        | Description                                                                                              |
+| --------------------- | ----------------------------------------------------------- | -------------------------------------------------------------------------------------------------------- |
+| id                    | snowflake                                                   | The ID of the referral                                                                                   |
+| user_id               | snowflake                                                   | The user who the referral was sent to                                                                    |
+| trial_id              | snowflake                                                   | The billing trial id                                                                                     |
```
Comment (dolfies) [inline pages/resources/premium-referral.mdx:15]:
```suggestion
| trial_id              | snowflake                                                   | The trial ID associated with the referral                                                                 |
```

File: pages/resources/premium-referral.mdx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+| subscription_trial    | billing subscription trial (todo)                           | The billing subscription for the the referral                                                            |
```
Comment (dolfies) [inline pages/resources/premium-referral.mdx:16]:
```suggestion
| subscription_trial    | subscription trial object (todo)                            | The subscription trial associated with the referral                                              |
```

File: pages/resources/premium-referral.mdx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+| expires_at            | ISO8601 timestamp                                           | The expiry date of the referral link                                                                     |
+| referrer_id           | snowflake                                                   | The user who created the referral                                                                        |
```
Comment (dolfies) [inline pages/resources/premium-referral.mdx:18]:
```suggestion
| referrer_id           | snowflake                                                   | The ID of the user who created the referral                                                           |
```

File: pages/resources/premium-referral.mdx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+
+### Premium Referral Eligibility Object
+
+Used to represent a user's premium referral eligibility.
+
```
Comment (dolfies) [inline pages/resources/premium-referral.mdx:24]: Unnecessary
```suggestion
```

File: pages/resources/premium-referral.mdx [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+###### Premium Referral Eligibility Structure
+| Field                 | Type                                                        | Description                                                                                              |
+| --------------------- | ----------------------------------------------------------- | -------------------------------------------------------------------------------------------------------- |
+| referrals_remaining   | number                                                      | Amount of referrals you have remaining                                                                   |
+| sent_user_ids         | array of user ids                                           | Users you have sent a referral to                                                                        |
```
Comment (dolfies) [inline pages/resources/premium-referral.mdx:29]:
```suggestion
| sent_user_ids         | array of user ids                                           | The users that have been referred                                                                        |
```

File: pages/resources/premium-referral.mdx [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+
+###### Example Premium Referral
```
Comment (dolfies) [inline pages/resources/premium-referral.mdx:32]: Please move this to be under the relevant structure

File: pages/resources/premium-referral.mdx [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+
+Returns a [premium referral](premium-referral-object). The referral is sent to the user id. Referrals expire after 48 hours. Only Nitro subscribers can create referrals. 
+
+<RouteHeader method="GET" url="/referrals/{referral.id}">
+  Get Premium Referral
+</RouteHeader>
+
+Returns a [premium referral](#premium-referral-object)
+
+<RouteHeader method="GET" url="/users/@me/referrals/eligibility">
```
Comment (dolfies) [inline pages/resources/premium-referral.mdx:76]: Please move this to the top of the endpoint list

**Review by dolfies — CHANGES_REQUESTED** [review]
Thanks for the PR! Just a few nitpicks. This new resource also needs to be added to the route header, but I can do that myself.

**Review by dolfies — APPROVED** [review]
ty!!

**Pull request merged**